src/sdl2_compat.c

```diff
@@ -5673,6 +5673,42 @@ SDL_GetDesktopDisplayMode(int displayIndex, SDL2_DisplayMode *mode)
     return 0;
 }
 
+#define PROP_WINDOW_FULLSCREEN_MODE "sdl2-compat.window.fullscreen-mode"
+
+static int
+ApplyFullscreenMode(SDL_Window *window)
+{
+    SDL_DisplayMode *mode = (SDL_DisplayMode *) SDL3_GetProperty(SDL3_GetWindowProperties(window), PROP_WINDOW_FULLSCREEN_MODE, NULL);
+    if (mode) {
+        /* Always try to enter fullscreen on the current display */
+        mode->displayID = SDL3_GetDisplayForWindow(window);
```

Comment:
Not sure what we're supposed to do if the current display isn't compatible with whatever mode the application wanted (for example, they enumerated modes for a different monitor which isn't present on the current monitor). Since SDL2 doesn't have a `displayID` field in the mode, we're forced to guess which monitor they want to use.

We could do one (or a combination) of the following:
1. Use the display of the window at the time the modeset is happening
2. Use the display of the window at the time `SDL_SetWindowDisplayMode()` is called
3. Enumerate modes on all displays and try to find the monitor that the caller's mode came from (which is quite possibly not unique to a single monitor).

Right now I'm doing 1 because it seems like probably what developers will expect and avoids the possibilty of the window flying across to another monitor when it enters fullscreen mode (as can happen with 2 or 3).

Reply:
This seems fine. For SDL3 the mode includes the display so the behavior is that the monitor is the display in the mode that was set, but for SDL2 the current window monitor is probably expected.